openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -68,9 +68,12 @@ def convert_empty_to_None(value):
     def get_profile_image(self, obj):
         """ Returns metadata about a user's profile image. """
         data = {'has_image': obj.has_profile_image}
+        if obj.has_profile_image:
```

Comment:
I think that the non-HTTP python API should handle this logic. You should be able to ask for a user's profile image and get the correct URL whether it's the default or the user's custom image.